lib/wallet/src/Cardano/Api/Gen.hs

```diff
@@ -1057,65 +1058,56 @@ genCostModels = do
 genExecutionUnitPrices :: Gen ExecutionUnitPrices
 genExecutionUnitPrices = ExecutionUnitPrices <$> genRational <*> genRational
 
-genProtocolParameters :: Gen ProtocolParameters
-genProtocolParameters =
-  ProtocolParameters
-    <$> ((,) <$> genNat <*> genNat)
-    <*> (Just <$> genRational)
-    <*> liftArbitrary genPraosNonce
-    <*> genNat
-    <*> genNat
-    <*> genNat
-    <*> genNat
-    <*> genNat
-    <*> liftArbitrary genLovelace
-    <*> genLovelace
-    <*> genLovelace
-    <*> genLovelace
-    <*> genEpochNo
-    <*> genNat
-    <*> genRationalInt64
-    <*> genRational
-    <*> genRational
-    <*> liftArbitrary genLovelace
-    <*> genCostModels
-    <*> liftArbitrary genExecutionUnitPrices
-    <*> liftArbitrary genExecutionUnits
-    <*> liftArbitrary genExecutionUnits
-    <*> liftArbitrary genNat
-    <*> liftArbitrary genNat
-    <*> liftArbitrary genNat
-    <*> liftArbitrary genLovelace
-
-genProtocolParametersWithAlonzoScripts :: Gen ProtocolParameters
-genProtocolParametersWithAlonzoScripts =
-  ProtocolParameters
-    <$> ((,) <$> genNat <*> genNat)
-    <*> (Just <$> genRational)
-    <*> liftArbitrary genPraosNonce
-    <*> genNat
-    <*> genNat
-    <*> genNat
-    <*> genNat
-    <*> genNat
-    <*> liftArbitrary genLovelace
-    <*> genLovelace
-    <*> genLovelace
-    <*> genLovelace
-    <*> genEpochNo
-    <*> genNat
-    <*> genRationalInt64
-    <*> genRational
-    <*> genRational
-    <*> (Just <$> genLovelace)
-    <*> genCostModels
-    <*> (Just <$> genExecutionUnitPrices)
-    <*> (Just <$> genExecutionUnits)
-    <*> (Just <$> genExecutionUnits)
-    <*> (Just <$> genNat)
-    <*> (Just <$> genNat)
-    <*> (Just <$> genNat)
-    <*> (Just <$> genLovelace)
+-- | A generator which only selects between two values for the sake of being
+-- fast. This is useful for generating values for @Cardano.TxBodyContent@,
+-- which only purpose is being hashed to produce part of the script integrity
+-- hash in the transaction.
+genProtocolParametersForHashing :: Gen ProtocolParameters
```

Comment:
Hi @Anviking 

I've created a PR that selects from a larger range of protocol parameters. On my system, it doesn't result in any significant slowdown. But how about on your system? (https://github.com/input-output-hk/cardano-wallet/pull/3935)

Reply:
I see a 10% slowdown (14-16s to 17s) with https://github.com/input-output-hk/cardano-wallet/pull/3935.

This made me realise, I should probably even have gone for a single value rather than 2 even.

7b92a3baf565f184549a809f23e61059945cfce4

This is the only thing it's used for: https://github.com/input-output-hk/cardano-ledger/blob/f49879a79098d9372d63baa13b94a941a56eda34/eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Tx.hs#L270-L280, and the hash will vary even with the same PParams.

With this I get around 11s instead.

To check how it scales I tried `withMaxSuccess 3_000` on `prop_balanceTransactionValid`; perhaps the slowdown was constant. But it was bigger than I'd expected: 53.6s with 50 PParams and 34s with 1.

I really don't think the pre-image of the script integrity hash deserves more than the reasonably minimal amount of CPU cycles.